File: enum_lookup_java/src/main/java/com/github/dkurata38/enum_lookup_java/after/RegistrationStatus.java
package com.github.dkurata38.enum_lookup_java.after;

public enum RegistrationStatus implements EnumLookupable {
    TEMPORARY("0"), REGULAR("1");

    private String code;

    RegistrationStatus(String code) {
        this.code = code;
    }

    public String getCode() {
        return code;
    }

    // 呼び出し側は, InterfaceのstaticメソッドをCallするだけ. 引数にEnumの検索条件を無名関数で指定.
    public static final EnumLookup<RegistrationStatus> byCode =
            new EnumLookup<>(RegistrationStatus.class, RegistrationStatus::getCode);
}

File: enum_lookup_scala/src/main/scala/com/github/dkurata38/enum_lookup_scala/RegistrationStatus.scala
package com.github.dkurata38.enum_lookup_scala

sealed class RegistrationStatus(code: String) extends ConstantSearchableByCode[RegistrationStatus](code)

object RegistrationStatus extends EnumSearchableByCode[RegistrationStatus] {
  case object Temporary extends RegistrationStatus("0")
  case object Regular extends RegistrationStatus("1")

  override def values: Seq[RegistrationStatus] = Seq(Temporary, Regular)
}